contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -76,10 +76,12 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
     uint256 public constant REQUIRED_ORIGIN_CHAIN_ID_START_INDEX = 4;
     uint256 public constant MINIMUM_REQUIRED_CALL_DATA_LENGTH = 68;
 
-    // The blockchain ID of the chain the contract is deployed on. Determined by warp messenger precompile.
-    bytes32 public immutable blockchainID;
+    // The blockchain ID of the chain the contract is deployed on.
+    // Set at most once by calling initializeBlockchainID() or receiveCrossChainMessage().
+    bytes32 public blockchainID = bytes32(0);
```

Comment:
do you need to explicity declare that it's `= bytes32(0)`, won't it be set for the default value?

If needed explicity, how come you decide to set in declaration here vs in constructor? Even if it were to be set in constructor and transaction runs out of gas, the nonce I don't think gets used up and can be deployed again. 
Was looking at https://ethereum.stackexchange.com/questions/114287/initialize-state-variables-in-the-contract-constructor-or-directly-in-the-declar, and seems result would be generally same, except more clear when used with derived contracts.

Reply:
Yeah you're correct that it will default to the zero value without being explicitly set, so I removed that now.

One clarification though, if a transaction runs out of gas it still is included in a block and will use the nonce of the account that sent it. Transactions that don't have enough of the native token to cover the maximum gas it could possibly use (ie `gasFeeCap * gasLimit`), aren't valid to be included in blocks, but whether or not the `gasLimit` is too low for the code that it calls is not known until the transaction itself is executed.

Reply:
ah yup my previous statement about nonce not being used up for gas running out was wrong